datalad/distribution/get.py

```diff
@@ -195,6 +195,7 @@ def _get_flexible_source_candidates_for_submodule(ds, sm):
         for k, v in sm.items()
         if k.startswith('gitmodule_')
     }
+    sm_candidate_props['basename'] = op.basename(sm_candidate_props["path"])
```

Comment:
not that familiar with code/logic here:

- is `path` 100% guaranteed to exist?
- is `basename` guaranteed to not be defined already? anyways, sounds better to be `path_basename` to relate to `path`

would need
- a test to demonstrate/test
- tune up to documentation to reflect is availability